File: packages/python-sdk/src/blindfold/regex/detectors/es/nie.py
"""Spanish NIE (Numero de Identidad de Extranjero) detector"""

import re

from ...base import RegexDetector
from ...entities import EntityType
from ...registry import register_region
from ...validators import es_nie_letter


@register_region("es")
class NieDetector(RegexDetector):
    entity_type = EntityType.ES_NIE
    score = 0.90
    context_keywords = [
        "nie", "extranjero", "numero de identidad",
    ]
    context_required = False
    context_window = 50

    # X/Y/Z + 7 digits + letter
    pattern = re.compile(
        r"\b[XYZ]\d{7}[-\s]?[A-Z]\b",
        re.IGNORECASE,
    )
    validator = staticmethod(es_nie_letter)

File: packages/js-sdk/src/regex/detectors/es/nie.ts
// Spanish NIE (Numero de Identidad de Extranjero) detector

import { RegexDetector } from '../../base'
import { EntityType } from '../../entities'
import { registerRegion } from '../../registry'
import { esNieLetter } from '../../validators'

class EsNieDetector extends RegexDetector {
  entityType = EntityType.ES_NIE
  score = 0.9
  contextKeywords = ['nie', 'extranjero', 'numero de identidad']
  contextRequired = false

  pattern = /\b[XYZ]\d{7}[-\s]?[A-Z]\b/gi
  validator = esNieLetter
}

registerRegion('es', EsNieDetector)

export { EsNieDetector }
